pkg/service/system/route53.go

```diff
@@ -129,3 +182,179 @@ func AddRoute53Record(ctx context.Context, dnsName, recordName, regionName strin
+		getRecordResult, err := route53Client.ListResourceRecordSetsWithContext(ctx, getRecordsInput)
+		if err != nil {
+			return nil, errors.Wrap(err, "GetRoute53Records: failed to get route53 records")
+		}
+
+		recordSets = append(recordSets, getRecordResult.ResourceRecordSets...)
+		if *getRecordResult.IsTruncated {
+			getRecordsInput.StartRecordName = getRecordResult.NextRecordName
+			getRecordsInput.StartRecordType = getRecordResult.NextRecordType
+			getRecordsInput.StartRecordIdentifier = getRecordResult.NextRecordIdentifier
+		} else {
+			break
+		}
+	}
+
+	for _, recordSet := range recordSets {
+		// filtering txt records only
+		if *recordSet.Type != route53.RRTypeTxt {
+			continue
+		}
+
+		for _, resourceRecord := range recordSet.ResourceRecords {
+			record := types.Route53Record{
+				Name:         *recordSet.Name,
+				Value:        *resourceRecord.Value,
+				Type:         *recordSet.Type,
+				TTLInSeconds: int64(*recordSet.TTL),
+			}
+
+			records = append(records, record)
+		}
+	}
+
+	return records, nil
+
+}
+
+// AppendRoute53Records adds records to the zone. It returns the records that were added.
+func AppendRoute53Records(ctx context.Context, records []types.Route53Record) error {
+
+	hostedZoneId := config.Get().AwsRoute53HostedZoneID
+
+	// Creating AWS Route53 session
+	route53Client, err := getRoute53Sess(emptyRegion)
+	if err != nil {
+		return errors.Wrap(err, "AddRoute53Record: failed to create route53 session")
+	}
+
+	hostedZone, err := getHostedZoneFromZoneId(route53Client, hostedZoneId)
+
+	if err != nil {
+		return errors.Wrap(err, "AddRoute53Record: failed to get route53 hostedZone ")
+	}
+
+	createInput := &route53.ChangeResourceRecordSetsInput{
+		ChangeBatch: &route53.ChangeBatch{
+			Changes: make([]*route53.Change, 0, len(records)),
+		},
+		HostedZoneId: aws.String(hostedZoneId),
+	}
+
+	for _, record := range records {
+
+		// AWS Route53 TXT record value must be enclosed in quotation marks on create
+		if record.Type == route53.RRTypeTxt {
+			record.Value = strconv.Quote(record.Value)
+		}
+
+		changeRequest := &route53.Change{
+			Action: aws.String("CREATE"),
+			ResourceRecordSet: &route53.ResourceRecordSet{
+				Name: aws.String(absoluteName(record.Name, hostedZone)),
+				ResourceRecords: []*route53.ResourceRecord{
+					{
+						Value: aws.String(record.Value),
+					},
+				},
+				TTL:  aws.Int64(int64(record.TTLInSeconds)),
+				Type: aws.String(record.Type),
+			},
+		}
+
+		createInput.ChangeBatch.Changes = append(createInput.ChangeBatch.Changes, changeRequest)
+
+	}
+
+	err = applyChange(ctx, route53Client, createInput)
+
+	if err != nil {
+		return errors.Wrap(err, "AppendRoute53Records: failed to create record")
+	}
+
+	return nil
+}
+
+// DeleteRoute53Records deletes the records from the zone. If a record does not have an ID,
+// it will be looked up. It returns the records that were deleted.
+func DeleteRoute53Records(ctx context.Context, records []types.Route53Record) error {
```

Comment:
This body of this method is almost entirely same as AppendRecord except for the `Action` can we refactor to pass action to a common method which does the rest?

Reply:
I think this is better as we anyway have a different method for "Get", If we do as you are saying, we have to rename the function as "(Change/Modify)Route53Records" with the action,  I think the one we have is better in readability, what do you think?

Reply:
We can get both readability and clean code by exposing 2 endpoints to Add and Delete which internally call Change with action. Currently is a lot of repeated code.

Reply:
Internally, in package I am calling this for both: 

func applyChange(ctx context.Context, route53Client *route53.Route53, input *route53.ChangeResourceRecordSetsInput) error {

Reply:
Updated